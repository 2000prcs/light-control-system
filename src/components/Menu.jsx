import React from 'react';
import { Button } from 'rendition';

const Display = () => {
  return (
    <div className="menu">
      <Button m={2} tertiary square>
        <i className="fas fa-home" />
      </Button>
      <Button w={80} m={2} tertiary>
        <i className="fas fa-angle-left" />
        <span>
          Back
        </span>
      </Button>
      <span className="lighting">
        <i className="far fa-lightbulb" />
        <span>
          Lighting
        </span>
      </span>
    </div>
  );
};

export default Display;