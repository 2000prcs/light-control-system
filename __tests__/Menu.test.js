import React from 'react';
import Menu from '../src/components/Menu.jsx';

let roomData = {
  active: false,
  brightness: 50,
  id: 1,
  name: 'Balcony',
};

describe('Menu component', () => {
  test('Should render Menu', () => {
    const MenuWrapper = shallow(<Menu room={roomData} />);
    expect(MenuWrapper).toMatchSnapshot();
  });
});
