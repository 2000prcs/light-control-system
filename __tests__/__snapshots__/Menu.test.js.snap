// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Should render Menu 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithTheme(Component)
          m={2}
          square={true}
          tertiary={true}
        >
          <i
            className="fas fa-home"
          />
        </WithTheme(Component)>,
        <WithTheme(Component)
          m={2}
          tertiary={true}
          w={80}
        >
          <i
            className="fas fa-angle-left"
          />
          <span>
            Back
          </span>
        </WithTheme(Component)>,
        <span
          className="lighting"
        >
          <i
            className="far fa-lightbulb"
          />
          <span>
            Lighting
          </span>
        </span>,
      ],
      "className": "menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <i
            className="fas fa-home"
          />,
          "m": 2,
          "square": true,
          "tertiary": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-home",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-angle-left"
            />,
            <span>
              Back
            </span>,
          ],
          "m": 2,
          "tertiary": true,
          "w": 80,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-angle-left",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back",
            },
            "ref": null,
            "rendered": "Back",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="far fa-lightbulb"
            />,
            <span>
              Lighting
            </span>,
          ],
          "className": "lighting",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "far fa-lightbulb",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lighting",
            },
            "ref": null,
            "rendered": "Lighting",
            "type": "span",
          },
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithTheme(Component)
            m={2}
            square={true}
            tertiary={true}
          >
            <i
              className="fas fa-home"
            />
          </WithTheme(Component)>,
          <WithTheme(Component)
            m={2}
            tertiary={true}
            w={80}
          >
            <i
              className="fas fa-angle-left"
            />
            <span>
              Back
            </span>
          </WithTheme(Component)>,
          <span
            className="lighting"
          >
            <i
              className="far fa-lightbulb"
            />
            <span>
              Lighting
            </span>
          </span>,
        ],
        "className": "menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <i
              className="fas fa-home"
            />,
            "m": 2,
            "square": true,
            "tertiary": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-home",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-angle-left"
              />,
              <span>
                Back
              </span>,
            ],
            "m": 2,
            "tertiary": true,
            "w": 80,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-angle-left",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back",
              },
              "ref": null,
              "rendered": "Back",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="far fa-lightbulb"
              />,
              <span>
                Lighting
              </span>,
            ],
            "className": "lighting",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "far fa-lightbulb",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lighting",
              },
              "ref": null,
              "rendered": "Lighting",
              "type": "span",
            },
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
